src/NHibernate/Linq/LinqExtensionMethods.cs

```diff
@@ -2442,7 +2442,7 @@ public static IFutureValue<TSource> ToFutureValue<TSource>(this IQueryable<TSour
 		{
 			var provider = GetNhProvider(source);
 			var future = provider.ExecuteFuture<TSource>(source.Expression);
-			return new FutureValue<TSource>(() => future, async cancellationToken => await future.GetEnumerableAsync(cancellationToken).ConfigureAwait(false));
+			return new FutureValue<TSource>(future.GetEnumerable, async cancellationToken => await future.GetEnumerableAsync(cancellationToken).ConfigureAwait(false));
```

Comment:
Is there a reason for not simply writing 
```c#
return new FutureValue<TSource>(future.GetEnumerable, future.GetEnumerableAsync);
```
instead for this line?

Reply:
Not really, I think before the return types were different (`IList<T>` -> `IEnumerable<T>`) so this was needed.

Reply:
Where is there an `IList<T>`? Both are `Task<IEnumerable<T>>`. (The method argument return type and `GetEnumerableAsync`.) So if there is no other reasons, I simplify it.

Reply:
My bad, [`IAsyncEnumerable<>`](https://github.com/nhibernate/nhibernate-core/pull/588/commits/3191b05bf7c7f76a5b7b8df7e4e385042619dc9a#diff-13fe75fe3520cc2ce957f347f2947980L2503) was the reason why as we had an anonymous funtion where we called the `ToList` method to get an `IEnumerable<>`. Now this is not needed anymore so the simplification is the way to go.